synapse/storage/events.py

```diff
@@ -374,6 +374,12 @@ def _persist_events(self, events_and_contexts, backfilled=False,
                     new_forward_extremeties=new_forward_extremeties,
                 )
                 persist_event_counter.inc_by(len(chunk))
+
+                for room_id, (_, _, new_state) in current_state_for_room.iteritems():
+                    self.get_current_state_ids.prefill(
+                        (room_id, ), new_state
+                    )
```

Comment:
Actually, I think it happens for free since we make the final dict out of state dicts that already have that conversion applied.